healsparse/healSparseMap.py

```diff
@@ -71,6 +71,11 @@ def __init__(self, cov_map=None, cov_index_map=None, sparse_map=None, nside_spar
             # this is a healpix_map input
             if sentinel is None:
                 sentinel = hp.UNSEEN
+            if is_integer_value(healpix_map[0]) and not is_integer_value(sentinel):
+                raise ValueError("The sentinel must be set to an integer value with an integer healpix_map")
+            elif not is_integer_value(healpix_map[0]) and is_integer_value(sentinel):
```

Comment:
do we need the second check? would python cast it to a float?

Reply:
I've been fairly strict with type checking in the whole code, and I'd like to keep it that way.  But while the harm in allowing implicit conversions from integer to float are also not large (as opposed to the other way), there are certainly cases where I've seen confusion because large integers cannot be converted to float and back.

Reply:
Thanks! It makes sense.